### DISCUSSION TITLE
"Security Risk" overview: inclusion of archived repositories

### DISCUSSION THREAD

**Discussion by SafeEval** [community/community#41158] [Code Security] [answered]
### Select Topic Area

Product Feedback

### Body

In the "Security Risk" overview (`https://github.com/orgs/<org>/security/risk`) for Secret Scanning, it shows the number of affected repositories. This is good. However, it does not include archived repositories with hardcoded secrets.

Unlike vulnerabilities in 1st party code and 3rd party dependencies, hardcoded secrets pose a risk just by existing in a repository, regardless of whether anything is deployed. If I conduct a sweep to remove hardcoded secrets from repositories in my organization, the dashboard might erroneously tell me that I have none remaining, when in truth there are live secrets lurking in archived repositories.

A helpful addition would be a toggle ("user's choice"), or a default for including archived repositories in the overview count for hardcoded secrets.

**Comment by obvioussean** [comment, marked as answer]
👋 Thank you for the feedback @SafeEval! With the release of the Risk and Coverage views we added the behavior for the metrics displayed to reflect the current filters. We had strong feedback that most of the time security engineers aren't concerned about archived repos, so we removed it from the default view. You can now choose the `Archived` view to see the counts and metrics for any archived repos.

![image](https://user-images.githubusercontent.com/15038374/207937063-179ab804-a798-49a1-a5be-cfa48550fc4d.png)

**Comment by SafeEval** [comment]
Thanks for the response. I agree with that archived repositories should be hidden from the overview, just wanted to point out that hardcoded secrets are different in this regard. I'm able to surface the hardcoded secrets in archived repositories, so this is workable. I'll still need to instruct my coworkers to look in two places when sweeping, but this will (hopefully) be a one-time thing for archived repositories.
